src/analysis/bounded_variation.lean

```diff
@@ -634,6 +635,147 @@ lemma monotone_on.has_locally_bounded_variation_on {f : α → ℝ} {s : set α}
   has_locally_bounded_variation_on f s :=
 λ a b as bs, ((hf.evariation_on_le as bs).trans_lt ennreal.of_real_lt_top).ne
 
+section variation_on_from_to
+
+
+/--
+The **signed** variation of `f` on the interval `Icc a b` intersected with the set `s`,
+squashed to a real (therefore only really meaningful if the variation is finite)
+-/
+noncomputable def variation_on_from_to (f : α → E) (s : set α) (a b : α) : real :=
+if a ≤ b then (evariation_on f (s ∩ Icc a b)).to_real else
+            - (evariation_on f (s ∩ Icc b a)).to_real
+
+variables {f : α → E} {s : set α}
```

Comment:
In most lemmas below, `f` and `s` should be explicit (unless there is a way to get them from another assumption).